humphrey/src/thread/pool.rs

```diff
@@ -104,8 +105,12 @@ impl ThreadPool {
 
     /// Stops the thread pool.
     pub fn stop(&mut self) {
-        self.recovery_thread = None;
-        self.tx.send(Message::Shutdown).unwrap();
+        self.tx = None;
```

Comment:
I'm probably missing something obvious, but if we're no longer sending `Message::Shutdown`, what breaks the threads out of their loops?

Reply:
I was referring to the regular (not recovery threads). I don't think `ThreadPool` was really supposed to be public - honestly this whole part of the project is very dodgy.

Reply:
It's not fresh on my mind, but I don't think there was a problem with regular threads. The drop impl handled it. In this WIP I changed it to be join so it would wait for them to exit for the memory checkers, but this might not be great behavior. As I mentioned in the OP, the `keep-alive` header makes this join take a long time
https://github.com/w-henderson/Humphrey/pull/114/files#diff-70f90e648844ce54d1a7dde834fbaa03af4128e594489fd77631137e852f88e8R208

Reply:
https://github.com/Grinkers/Humphrey/blob/8324254edc2f3f4ca6973d93d71a4762b3fa3f4a/humphrey/src/thread/pool.rs#L162
I later realized I didn't really answer the question, regular threads also break with a similar pattern. I made a small push to have an explicit send too, although not strictly needed.

Reply:
Okay, that makes sense - can you explain why `self.tx` needs to be an `Option`?